### ISSUE TITLE
log4j:WARN Please initialize the log4j system properly

### ISSUE THREAD

**Issue by robin820224** [elastic/logstash#4704]
I have installed logstash2.2.2 and elasticsearch2.1.1 in centos7. When I used "service logstash configtest", I have got log4j WARM:

[root@iZ25g5gvvtgZ bin]# service logstash configtest
log4j:WARN No appenders could be found for logger (org.apache.http.client.protocol.RequestAuthCache).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Configuration OK

The follow is my conf file:
input {
  log4j {
    mode => "server"
    host => "localhost"
    port => 4561
  }
}
output {
  elasticsearch {
    hosts => "localhost:9200"
  }
}

**Comment by jordansissel** [comment]
You can ignore this warning; it is meaningless.

We will make it go away in a future release.

**Comment by robin820224** [comment]
After I start logstash service, it seems that logstash had listened on tcp 4561 port. I want to known whether logstash could support UDP protocal? If can, how should I config the conf file for udp protocal?

[root@iZ25g5gvvtgZ conf.d]# netstat -ant | grep 4561
tcp        0      0 0.0.0.0:4561            0.0.0.0:*               LISTEN

**Comment by zhoufwind** [comment]
Hi folks, try to upgrade Logstash to 2.2.2 from 2.1.1 today, received same message.
![2016-02-24-01-logstash-2 2 2-kafka-latency-05](https://cloud.githubusercontent.com/assets/3394857/13278294/73893cea-db0a-11e5-9717-65a25bb8963c.PNG)
Regardless these messages, I have got another problem here: **Logstash take 4 minutes to startup.**
Normally, it only take seconds to complete the collection(read the log, transfer them to kafka broker).
But with version 2.2.2., it will take 2 minutes to read the logs, and spend another 2 minutes to transfer them to kafka broker, pasting test time stamp as below:
02:42PM - Stop logstash service and remove sincedb file manually
02:44PM - Start logstash service
02:46PM - Logstash start to read logs
02:48PM - sincedb file been created, Logstash transfer the logs to Kafka broker(logs transfer to ES immediately, we can check them on Kibana)
![2016-02-24-01-logstash-2 2 2-kafka-latency-03](https://cloud.githubusercontent.com/assets/3394857/13278542/40f48e68-db0c-11e5-9ad4-93c2ca05c086.PNG)
02:54PM - Stop logstash service, Kafka log throw exception as below:
![2016-02-24-01-logstash-2 2 2-kafka-latency-04](https://cloud.githubusercontent.com/assets/3394857/13278702/6c7c2cf2-db0d-11e5-9942-afbdce77e55f.PNG)

Not sure how to debug the root cause further, wondering if anyone got same problem like me.
Considering this unstable factor, I will use current version till stable version been released.

**Comment by andrewvc** [comment]
@zhoufwind is this issue only with the Kafka broker? Does logstash startup quickly with other configs? That info would help us greatly. If it is a Kafka issue could report the bug here: https://github.com/logstash-plugins/logstash-input-kafka

**Comment by zhoufwind** [comment]
@andrewvc sorry for late responding on it.
Good news is I cannot reproduce same issue anymore when I try to debug on other machines(both Linux and Windows) today, even origin one(10.10.10.66) startup quickly as well.
One possible symptom is: When I stop or restart logstash service, java process didn't been killed actually. After I killed logstash & java process manually, upgrade logstash latest version(2.2.2 from 2.1.1), start logstash process again, everything working perfect then.
So, Logstash-2.2.2 is working perfect right now and turns out it's my personal problem, going on update them next week and thanks all for taking care of my mistaken, sincerely.

**Comment by glmrenard** [comment]
Looks like #4111
_cat /var/log/logstash/logstash.err_
```
log4j:WARN No appenders could be found for logger (io.netty.util.internal.logging.InternalLoggerFactory).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
```

With logtash 2.4.0, jdk 1.8.0_101 and centos 7.2

**Comment by HarriL** [comment]
@andrewvc 

Got these WARN's:
> Jun 23 20:48:32 localhost.localdomain logstash[78339]: log4j:WARN No appenders could be found for logger (org.apache.http.client.protocol.RequestAuthCache).
> Jun 23 20:48:32 localhost.localdomain logstash[78339]: log4j:WARN Please initialize the log4j system properly.
> Jun 23 20:48:32 localhost.localdomain logstash[78339]: log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.

The issue with Java, ES, Logstash installation, not in Kibana (not yet istalled), nor Kafka (not installed). **Only installed is CentOS+ES+Java+Logstash.** 

# ENVIRONMENT
## OS: 
Linux localhost.localdomain 3.10.0-514.21.2.el7.x86_64 #1 SMP Tue Jun 20 12:24:47 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
I run the OS in VMWare Fusion w/ 2GB RAM, but only w/ one processor.

## JAVA:
Java: jre-8u131-linux-x64.rpm, openjdk version "1.8.0_131"
OpenJDK Runtime Environment (build 1.8.0_131-b12)
OpenJDK 64-Bit Server VM (build 25.131-b12, mixed mode)

## ELK:
elasticsearch.noarch 0:5.0.0_alpha5-1
logstash 5.4.2
But Kibana is not installed, nor Kafka, so as the OS I have only CentOS7 Graphical w/ latest updates. And ES+Logstash w/ Java. But not eg. log4j nor anything else!  

# Installation 
## Order
OS+updates, latest Java from Oracle, ES, Logstash

## ES installation:
`$ sudo rpm --import http://packages.elastic.co/GPG-KEY-elasticsearch`
`$ sudo vi /etc/yum.repos.d/elasticsearch.repo`
Inserted_ the following lines to elasticsearch.repo: 
[elasticsearch]
name=Elasticsearch repository
baseurl=http://packages.elastic.co/elasticsearch/5.x/centos
gpgcheck=1
gpgkey=http://packages.elastic.co/GPG-KEY-elasticsearch
enabled=1
`$ sudo yum install elasticsearch`
…
Installed:
  elasticsearch.noarch 0:5.0.0_alpha5-1
`$ sudo  curl -X GET http://localhost:9200`
{
  "name" : "eO9lReY",
  "cluster_name" : "elasticsearch",
  "version" : {
    "number" : "5.0.0-alpha5",
    "build_hash" : "d327dd4",
    "build_date" : "2016-08-04T08:59:39.568Z",
    "build_snapshot" : false,
    "lucene_version" : "6.1.0"
  },
  "tagline" : "You Know, for Search"
}

## Logstash installation:
`$ rpm --import https://artifacts.elastic.co/GPG-KEY-elasticsearch`
`$ vi /etc/yum.repos.d/logstash.repo`
_Inserted_ the following lines to the repository configuration file logstash.repo:
[logstash-5.x]
name=Elastic repository for 5.x packages
baseurl=https://artifacts.elastic.co/packages/5.x/yum
gpgcheck=1
gpgkey=https://artifacts.elastic.co/GPG-KEY-elasticsearch
enabled=1
autorefresh=1
type=rpm-md
`$ sudo yum install logstash`
`$ sudo systemctl enable logstash`
`$ sudo systemctl start logstash`

# Config and test
Then PKI...and config input, output and filter files for Logstash and full restart of the service+testing
`$ sudo systemctl daemon-reload`
`$ sudo systemctl restart logstash`
`$ sudo /usr/share/logstash/bin/logstash --config.test_and_exit --path.settings /etc/logstash`
OpenJDK 64-Bit Server VM warning: If the number of processors is expected to increase from one, then you should configure the number of parallel GC threads appropriately using -XX:ParallelGCThreads=N
Sending Logstash's logs to /var/log/logstash which is now configured via log4j2.properties
Configuration OK

# And then check the status
`$ sudo systemctl status logstash -l`
● logstash.service - logstash
   Loaded: loaded (/etc/systemd/system/logstash.service; enabled; vendor preset: disabled)
   Active: active (running) since Fri 2017-06-23 20:48:16 EEST; 15h ago
 Main PID: 78339 (java)
   CGroup: /system.slice/logstash.service
           └─78339 /usr/bin/java -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=75 -XX:+UseCMSInitiatingOccupancyOnly -XX:+DisableExplicitGC -Djava.awt.headless=true -Dfile.encoding=UTF-8 -XX:+HeapDumpOnOutOfMemoryError -Xmx1g -Xms256m -Xss2048k -Djffi.boot.library.path=/usr/share/logstash/vendor/jruby/lib/jni -Xbootclasspath/a:/usr/share/logstash/vendor/jruby/lib/jruby.jar -classpath : -Djruby.home=/usr/share/logstash/vendor/jruby -Djruby.lib=/usr/share/logstash/vendor/jruby/lib -Djruby.script=jruby -Djruby.shell=/bin/sh org.jruby.Main /usr/share/logstash/lib/bootstrap/environment.rb logstash/runner.rb --path.settings /etc/logstash

Jun 23 20:48:16 localhost.localdomain systemd[1]: Started logstash.
Jun 23 20:48:16 localhost.localdomain systemd[1]: Starting logstash...
Jun 23 20:48:17 localhost.localdomain logstash[78339]: OpenJDK 64-Bit Server VM warning: If the number of processors is expected to increase from one, then you should configure the number of parallel GC threads appropriately using -XX:ParallelGCThreads=N
Jun 23 20:48:30 localhost.localdomain logstash[78339]: Sending Logstash's logs to /var/log/logstash which is now configured via log4j2.properties
Jun 23 20:48:32 localhost.localdomain logstash[78339]: log4j:WARN No appenders could be found for logger (org.apache.http.client.protocol.RequestAuthCache).
Jun 23 20:48:32 localhost.localdomain logstash[78339]: log4j:WARN Please initialize the log4j system properly.
Jun 23 20:48:32 localhost.localdomain logstash[78339]: log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.

**So the WARN is there.**

# FYI conf files:
`$ sudo ls /var/log/logstash/logstash.err`
ls: cannot access /var/log/logstash/logstash.err: No such file or directory
`$ sudo ls /var/log/logstash`
logstash-plain-2017-06-22.log  logstash-plain-2017-06-23.log  logstash-plain.log

/etc/logstash/conf.d/input.conf
> input {
> beats {
> port => 5044
> ssl => true
> ssl_certificate => "/etc/pki/tls/certs/logstash-forwarder.crt"
> ssl_key => "/etc/pki/tls/private/logstash-forwarder.key"
> }
> }

/etc/logstash/conf.d/output.conf
> output {
> elasticsearch {
> hosts => ["localhost:9200"]
> sniffing => true
> manage_template => false
> index => "%{[@metadata][beat]}-%{+YYYY.MM.dd}"
> document_type => "%{[@metadata][type]}"
> }
> }

/etc/logstash/conf.d/filter.conf
> filter {
> if [type] == "syslog" {
> grok {
> match => { "message" => "%{SYSLOGLINE}" }
> }
> date {
> match => [ "timestamp", "MMM  d HH:mm:ss", "MMM dd HH:mm:ss" ]
> }
> }
> }

**Comment by jakelandis** [comment]
Assuming the original intent of this issue, I am closing this issue as resolved in 5.6.0 and 6.0.0

> log4j:WARN Please initialize the log4j system properly

and 

>You can ignore this warning; it is meaningless.

and

> We will make it go away in a future release.

That future release will be 5.6.0 and 6.0.0, resolved as part of https://github.com/elastic/logstash/issues/7526

**Comment by chandan188** [comment]
May be you have not kept the log4j.properties file inside src/main/resource folder.
